staging/src/k8s.io/mount-utils/mount_linux.go

```diff
@@ -562,6 +563,17 @@ func (mounter *SafeFormatAndMount) formatAndMountSensitive(source string, target
 	return nil
 }
 
+func (mounter *SafeFormatAndMount) format(fstype string, args []string) ([]byte, error) {
+	if mounter.formatSem != nil {
+		mounter.formatSem <- 1
+		defer func() {
+			<-mounter.formatSem
+		}()
+	}
+
+	return mounter.Exec.Command("mkfs."+fstype, args...).CombinedOutput()
```

Comment:
If for some reason the mkfs command hangs things could get stuck. I wonder if we want a timeout on waiting for the semaphore? We'll allow more than N concurrent mounts but only if they're getting stuck, in which case we'll OOM which will maybe unstick things. (and if it doesn't we're no worse off than we were with the stuck mkfs consuming the semaphores).

Reply:
Good idea. Done and tested.

Reply:
we should have used something like https://pkg.go.dev/golang.org/x/time/rate, that is easier to think aboutm and `CommandContext()` to kill the command on timeout